File: src/main/java/com/github/eendroroy/sdk/openweather/endpoint/WeatherEndpoints.java
package com.github.eendroroy.sdk.openweather.endpoint;

import com.github.eendroroy.sdk.openweather.response.FindWeatherResponse;
import com.github.eendroroy.sdk.openweather.response.GetWeatherResponse;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * @author indrajit
 */

public interface WeatherEndpoints {
    @GET("/data/2.5/weather")
    Call<GetWeatherResponse> weatherByCityName(
            @Query("q") String cityName,
            @Query("appid") String apiKey
    );

    @GET("/data/2.5/weather")
    Call<GetWeatherResponse> weatherByCityId(
            @Query("id") String cityId,
            @Query("appid") String apiKey
    );

    @GET("/data/2.5/weather")
    Call<GetWeatherResponse> weatherByCoOrd(
            @Query("lat") String latitude,
            @Query("lon") String longitude,
            @Query("appid") String apiKey
    );

    @GET("/data/2.5/weather")
    Call<GetWeatherResponse> weatherByZip(
            @Query("zip") String zipCode,
            @Query("appid") String apiKey
    );

    @GET("/data/2.5/find")
    Call<FindWeatherResponse> weatherByCityName(
            @Query("q") String cityName,
            @Query("appid") String apiKey,
            @Query("units") String unit
    );
}

File: src/main/kotlin/com/github/eendroroy/sdk/openweathermap/endpoint/WeatherEndpoints.kt
package com.github.eendroroy.sdk.openweathermap.endpoint

import com.github.eendroroy.sdk.openweathermap.response.FindWeatherResponse
import com.github.eendroroy.sdk.openweathermap.response.GetWeatherResponse
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * @author indrajit
 */
interface WeatherEndpoints {
    @GET("/data/2.5/weather")
    fun weatherByCityName(
            @Query("q") cityName: String?,
            @Query("appid") apiKey: String?
    ): Call<GetWeatherResponse?>

    @GET("/data/2.5/weather")
    fun weatherByCityId(
            @Query("id") cityId: String?,
            @Query("appid") apiKey: String?
    ): Call<GetWeatherResponse?>

    @GET("/data/2.5/weather")
    fun weatherByCoOrd(
            @Query("lat") latitude: String?,
            @Query("lon") longitude: String?,
            @Query("appid") apiKey: String?
    ): Call<GetWeatherResponse?>

    @GET("/data/2.5/weather")
    fun weatherByZip(
            @Query("zip") zipCode: String?,
            @Query("appid") apiKey: String?
    ): Call<GetWeatherResponse?>

    @GET("/data/2.5/find")
    fun weatherByCityName(
            @Query("q") cityName: String?,
            @Query("appid") apiKey: String?,
            @Query("units") unit: String?
    ): Call<FindWeatherResponse?>
}